logs/field.go

```diff
@@ -0,0 +1,321 @@
+package logs
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// Field represents typed log field (a key-value pair). Adapters should determine
+// Field's type based on Type and use the corresponding getter method to retrieve
+// the value:
+//
+//	switch f.Type() {
+//	case logs.IntType:
+//	    var i int = f.Int()
+//	    // handle int value
+//	case logs.StringType:
+//	    var s string = f.String()
+//	    // handle string value
+//	//...
+//	}
+//
+// Getter methods must not be called on fields with wrong Type (e.g. calling String()
+// on fields with Type != StringType).
+// Field must not be initialized directly as a struct literal.
+type Field struct {
+	ftype FieldType
+	key   string
+
+	vint int64
+	vstr string
+	vany interface{}
+}
+
+func (f Field) Type() FieldType {
+	return f.ftype
+}
+
+func (f Field) Key() string {
+	return f.key
+}
+
+// String is a value getter for fields with StringType type
+func (f Field) StringValue() string {
+	f.checkType(StringType)
+	return f.vstr
+}
+
+// Int is a value getter for fields with IntType type
+func (f Field) IntValue() int {
+	f.checkType(IntType)
+	return int(f.vint)
+}
+
+func (f Field) Int64Value() int64 {
+	f.checkType(Int64Type)
+	return f.vint
+}
+
+// Bool is a value getter for fields with BoolType type
+func (f Field) BoolValue() bool {
+	f.checkType(BoolType)
+	return f.vint != 0
+}
+
+// Duration is a value getter for fields with DurationType type
+func (f Field) DurationValue() time.Duration {
+	f.checkType(DurationType)
+	return time.Nanosecond * time.Duration(f.vint)
+}
+
+// Strings is a value getter for fields with StringsType type
+func (f Field) StringsValue() []string {
+	f.checkType(StringsType)
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.([]string)
+}
+
+// Error is a value getter for fields with ErrorType type
+func (f Field) ErrorValue() error {
+	f.checkType(ErrorType)
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.(error)
+}
+
+// Any is a value getter for fields with AnyType type
+func (f Field) AnyValue() interface{} {
```

Comment:
Need test for AnyValue()

Reply:
in this commit
https://github.com/ydb-platform/ydb-go-sdk/pull/579/commits/2ecccc30ae59b64f9116f0e4fea1827b77bc9447